packages/vscode-extension/src/webview/providers/UtilsProvider.tsx

```diff
@@ -17,12 +17,30 @@ export const Platform = {
   },
 };
 
+type UtilsContextProps = {
+  utils: UtilsInterface;
+  telemetryEnabled: boolean;
+};
+
 const utils = makeProxy<Utils>("Utils");
 
-const UtilsContext = createContext<UtilsInterface>(utils);
+const UtilsContext = createContext<UtilsContextProps>({ utils, telemetryEnabled: false });
 
 export function UtilsProvider({ children }: PropsWithChildren) {
-  return <UtilsContext.Provider value={utils}>{children}</UtilsContext.Provider>;
+  const [telemetryEnabled, setTelemetryEnabled] = useState(false);
+
+  useEffect(() => {
```

Comment:
I think it should be in a separate hook, as a change to the state will cause unnecessary rerenders in all useUtils occurrences

Reply:
Something like this will be better imo. 

```
const useTelemntry = () => {
  const utils = useUtils();
  const [telemetryEnabled, setTelemetryEnabled] = useState(false);

  useEffect(() => {
    utils.isTelemetryEnabled().then(setTelemetryEnabled);
    utils.addListener("telemetryEnabledChanged", setTelemetryEnabled);

    return () => {
      utils.removeListener("telemetryEnabledChanged", setTelemetryEnabled);
    };
  }, []);

  return telemetryEnabled;
}
```